Wrappers/Python/cil/optimisation/algorithms/Algorithm.py

```diff
@@ -31,167 +87,140 @@ class Algorithm(object):
       stop as soon as the stop criterion is met.
       The user is required to implement the :code:`set_up`, :code:`__init__`, :code:`update` and
       and :code:`update_objective` methods
-      
+
       A courtesy method :code:`run` is available to run :code:`n` iterations. The method accepts
-      a :code:`callback` function that receives the current iteration number and the actual objective
-      value and can be used to trigger print to screens and other user interactions. The :code:`run`
-      method will stop when the stopping criterion is met. 
+      a :code:`callbacks` list of callables, each of which receive the current Algorithm object
+      (which in turn contains the iteration number and the actual objective value)
+      and can be used to trigger print to screens and other user interactions. The :code:`run`
+      method will stop when the stopping criterion is met or `StopIteration` is raised.
    '''
 
-    def __init__(self, **kwargs):
-        '''Constructor
-        
-        Set the minimal number of parameters:
-        
-        
+    def __init__(self, max_iteration=0, update_objective_interval=1, **kwargs):
+        '''Set the minimal number of parameters:
+
         :param max_iteration: maximum number of iterations
         :type max_iteration: int, optional, default 0
         :param update_objective_interval: the interval every which we would save the current\
                                        objective. 1 means every iteration, 2 every 2 iteration\
                                        and so forth. This is by default 1 and should be increased\
                                        when evaluating the objective is computationally expensive.
         :type update_objective_interval: int, optional, default 1
-        :param log_file: log verbose output to file
-        :type log_file: str, optional, default None
         '''
         self.iteration = -1
-        self.__max_iteration = kwargs.get('max_iteration', 0)
+        self.__max_iteration = max_iteration
```

Comment:
I guess this is my bad, but it shouldn't be a class variable (double `_`), rather a class member.

Reply:
afaik the convention in this file does *not* use double `__` to mark class variables. afaik the convention here is:

Python prefix | equivalent C++ meaning
--|--
None | `public`
`_\w` | `protected`
`__` | `private`